### PR TITLE
Fix value type with static method fails to load

### LINKED ISSUE #73322: TypeEquivalence - value types with a `static` method shouldn't fail to load
If an assembly is marked with `PrimaryInteropAssemblyAttribute`, all public value types are validated for TypeEquivalence. This works without issue unless the value type contains a method, at which point a failure will occur:

```
  System.TypeLoadException: Could not load the structure '<TYPE_NAME>' from assembly '<ASSEMBLY_NAME>'. The structure is marked as eligible for type equivalence, but it has a method.
```

However, if the type only has methods that are marked `static`, this should be permitted.

Fails as expected:
```csharp
struct T
{
    public int F;
    public void M() { }
}
```

Consider supported:
```csharp
struct T
{
    public int F;
    public static void M() { }
}
```

### PR DESCRIPTION
**Pull request by JJamesWWang**
fixes #73322

Allows static methods if checking type equivalence when marked with `PrimaryInteropAssemblyAttribute`

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=76262) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=76262) before we can accept your contribution.<br/><br/>:x: JJamesWWang [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=76262)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=76262) it.</sub>

File: src/tests/baseservices/typeequivalence/simple/Simple.cs [hunk 1]
```diff
@@ -253,6 +254,13 @@ private static unsafe void TestTypeEquivalenceWithTypePunning()
         }
     }
 
+    [MethodImpl (MethodImplOptions.NoInlining)]
+    private static void TestLoadingValueTypeWithStaticMethod() 
+    {
+        Console.WriteLine($"{nameof(TestLoadingValueTypeWithStaticMethod)}");
+        Console.WriteLine($"-- {typeof(ValueTypeWithStaticMethod).Name}");
+    }
```
Comment (AaronRobinsonMSFT) [inline src/tests/baseservices/typeequivalence/simple/Simple.cs:262]: I think we should also have a regression based test. Let's attempt to load a type that will throw and validate the exception type is expected. Create a new type called `ValueTypeWithMethod` and use that to verify it throws as expected. We can use a local static function like below, which can be pasted within this method. Remember to update the name of `TestLoadingValueTypeWithStaticMethod`.

```csharp
Assert.Throws<TypeLoadException>(() => LoadInvalidType());

[MethodImpl (MethodImplOptions.NoInlining)]
static void LoadInvalidType()
{
        Console.WriteLine($"-- {typeof(ValueTypeWithMethod).Name}");
}
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**